test/illuminate/Illuminate.sol

```diff
@@ -2,48 +2,95 @@
-  /// @param r address of a deployed relayer contract to use
-  constructor(address m, address r) {
+  constructor() {
     admin = msg.sender;
   }
 
+  /// @param u underlying
+  /// @param m maturity
+  /// @param t principal token addresses for this market minus the illuminate principal (which is added here)
+  /// @param n name for the illuminate token
+  /// @param s symbol for the illuminate token
+  /// @param d decimals for the illuminate token
+  function createMarket(address u, uint256 m, address[7] memory t, string calldata n, string calldata s, uint8 d) external authorized(admin) returns (bool) {
+    require(markets[u][m][uint256(Principals.Illuminate)] == address(0), 'market exists'); 
+
+    // deploy an illuminate token with this new market NOTE: ATM is using name as symbol args
+    address iToken = address(new ZcToken(u, m, n, s, d));
+
+    // the market will have the illuminate principal as its zeroth item, thus t should have Principals[1] as [0]
+    // TODO we could choose to put illuminate last in
+    address[8] memory market = [iToken, t[0], t[1], t[2], t[3], t[4], t[5], t[6]];
+
+    uint256 max = 2**256 - 1;
+    // approve the underlying for max per given principal
+    for (uint8 i; i < 8; i++) {
+      Safe.approve(IErc20(t[i]), redeemer, max);
+    }
+
+    markets[u][m] = market;
+
+    // TODO "args" for this event?
+    emit CreateMarket(u, m);
+    // ...
+    return true;
+  }
+
   /// @dev mint is uniform across all principals, thus there is no need for a 'minter'
-  /// @param p value of a specific principal according to the MarketPlace Principals Enum. NOTE (if using a local Principals this type can change)
-  /// @param u address of an underlying asset
-  /// @param m maturity (timestamp) of the market
-  /// @param a amount being minted
-  function mint(uint8 p, address u, uint256 m, uint256 a) public returns (bool) {
-    address mPlace = marketPlace(marketPlace);
+  // @param p value of a specific principal according to the MarketPlace Principals Enum.
+  // @param u address of an underlying asset
+  // @param m maturity (timestamp) of the market
+  // @param a amount being minted
+  // TODO move to lender.sol
+  // function mint(uint8 p, address u, uint256 m, uint256 a) public returns (bool) {
+    // address mPlace = marketPlace(marketPlace);
     // use market interface to fetch the market for the given market pair
-    address[8] market = mPlace.markets(u, m);
+    // address[8] market = mPlace.markets(u, m);
     // use safe transfer lib and ERC interface...
-    Safe.transferFrom(Erc20(market[p]), msg.sender, address(this), a);
+    // Safe.transferFrom(Erc20(market[p]), msg.sender, address(this), a);
     // use zctoken interface...
-    ZcToken(market[mPlace.Principals.Illuminate]).mint(msg.sender, a);
+    // ZcToken(market[mPlace.Principals.Illuminate]).mint(msg.sender, a);
 
-    emit Mint(p, u, m, a);
+    // emit Mint(p, u, m, a);
 
-    return true;
+    // return true;
+  // }
+
```

Comment:
Remove commented out code.

Reply:
LOL. yes. but. you are actually going to do it. why? this `mint` method is going to be moved into the `lender`

just copy it from here.